bpf/process/types/basic.h

```diff
@@ -771,20 +771,40 @@ struct ip_ver {
 // use the selector value to determine a LPM Trie map, and do a lookup to determine whether the argument
 // is in the defined set.
 static inline __attribute__((always_inline)) long
-filter_addr4_map(struct selector_arg_filter *filter, __u32 addr)
+filter_addr_map(struct selector_arg_filter *filter, __u64 *addr, __u16 family)
 {
 	void *addrmap;
-	__u32 map_idx = filter->value;
-	struct addr4_lpm_trie arg;
-
-	addrmap = map_lookup_elem(&addr4lpm_maps, &map_idx);
-	if (!addrmap)
+	__u32 *map_idxs = (__u32 *)&filter->value;
+	__u32 map_idx;
+	struct addr4_lpm_trie arg4;
+	struct addr6_lpm_trie arg6;
+	void *arg;
+
+	switch (family) {
+	case AF_INET:
+		map_idx = map_idxs[0];
+		addrmap = map_lookup_elem(&addr4lpm_maps, &map_idx);
+		if (!addrmap)
+			return 0;
+		arg4.prefix = 32;
+		arg4.addr = addr[0];
+		arg = &arg4;
+		break;
+	case AF_INET6:
+		map_idx = map_idxs[1];
+		addrmap = map_lookup_elem(&addr6lpm_maps, &map_idx);
+		if (!addrmap)
+			return 0;
+		arg6.prefix = 128;
```

Comment:
Out of curiosity, what is the reason for using an lpm tree if we are not matching prefixes. Do we plan to support matching besides /32, or just to save space?

Reply:
We are matching prefixes! The user can specify CIDRs with /nn netmask lengths and these are entered into the map. We then observe an IP address and ask for the longest prefix that matches it, specifying how many bits we have observed, which is the whole of the IP address so 32 bits for IPv4 and 128 bits for IPv6.
For example, we could store 127.0.0.0/8 in the map, and then observe 127.0.0.1 as an IP address. This will return the 127.0.0.0/8 match.

Reply:
Thanks!